buildLogic/toolchain-convention/src/main/kotlin/app/cash/sqldelight/toolchain/ToolchainConventions.kt

```diff
@@ -1,47 +1,22 @@
 package app.cash.sqldelight.toolchain
 
-import org.gradle.api.JavaVersion
 import org.gradle.api.Plugin
 import org.gradle.api.Project
-import org.gradle.api.plugins.JavaPluginExtension
 import org.gradle.jvm.toolchain.JavaLanguageVersion
 import org.gradle.jvm.toolchain.JvmVendorSpec
-import org.jetbrains.kotlin.gradle.dsl.JvmTarget
 import org.jetbrains.kotlin.gradle.dsl.kotlinExtension
-import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
-import com.android.build.gradle.BaseExtension as AndroidBaseExtension
 
-abstract class ToolchainConventions(private val targetJdkVersion: String) : Plugin<Project> {
+abstract class ToolchainConventions(private val jdkVersion: Int) : Plugin<Project> {
   override fun apply(project: Project) {
     project.kotlinExtension.jvmToolchain { spec ->
-      spec.languageVersion.set(JavaLanguageVersion.of(BUILD_JDK))
+      spec.languageVersion.set(JavaLanguageVersion.of(jdkVersion))
```

Comment:
It's not clear to me why this change is needed. There are still some benefits to using JDK 17 (or newer) as the toolchain while setting target/source compatibility separately.

Reply:
I am okay with both implementations. My main reason to change it was to use the agp-api artifact in the future, which removed `AndroidBaseExtension` and I got no IDE support due to failing builds. If we want to use newer JDKs, I will revert it.

Reply:
We could probably use `CommonExtension<*, *, *, *>` instead which is part of AGP's public API. I would otherwise prefer leaving the previous toolchain setup.

Reply:
You are right, the AGP (impl) already contains an api package with the `CommonExtension`.